### PR TITLE
ARGO-154 Establish tenant db configuration options

### PR DESCRIPTION
**Pull request by kaggis**
This implementation allows CE to retrieve tenant_db_information in order to establish the destinations (server/port/db) for the results to be written to.

- In the future CE will retrieve tenant_db info from the api defined information in the datastore. Up until then the tenant_db information is established as an additional configuration file (residing in /etc/ar-compute/) per tenant.

- The utils.ArgoConfiguration which is responsible for holding the CE enviroment configuration has now a new method to load tenant_db_information based on the tenant name
 - also has some handy methods to return db_conf information in various forms(mongo uri, mongo host string, mongo port int etc...)
- job submittion scripts (job_ar.py, job_status_detailed.py) now use the utils.ArgoConfiguration object and thus retrieve tenant based mongo config

- unittests included

### CODE REVIEW

**Comment by kaggis** [comment]
Now opened against the multitenant branch

please review 
/cc @dpavlos @pkoro

File: bin/utils.py [hunk 1]
```diff
@@ -56,10 +67,46 @@ def load_config(self, filename):
...
-        self.sync_exec = ar_config.get('connectors','sync_exec')
-        self.sync_path = ar_config.get('connectors','sync_path')
+        self.sync_exec = ar_config.get('connectors', 'sync_exec')
+        self.sync_path = ar_config.get('connectors', 'sync_path')
+
+        # Grab run mode
+        self.mode = ar_config.get('default', 'mode')
+
+        # Grab mapping info (will be removed in the near future)
+        self.n_alt = ar_config.get('connectors', 'n_alt')
```
Comment (pkoro) [inline bin/utils.py]: tests break here. I think you should remove lines 79-85 (`connectors` and `sampling` sections) or we should add them in `utils_test.py`.

File: bin/utils_test.py [hunk 2]
```diff
@@ -21,6 +21,42 @@
...
+"""
+
+TEST_TENANT_DB_CONTENTS = r"""
+{
+  "db_conf": [
+    {
+      "store": "ar",
+      "server": "192.168.0.99",
+      "port": 27017,
+      "database": "argo-egi",
```
Comment (pkoro) [inline bin/utils_test.py]: Better change `argo-egi` to `argo_EGI` (the same name we have it on test data on devel API node).

File: bin/utils_test.py [hunk 3]
```diff
@@ -21,6 +21,42 @@
...
+      "username": "",
+      "password": ""
+    },
+    {
+      "store": "status",
+      "server": "192.168.0.99",
+      "port": 27017,
+      "database": "argo-egi",
```
Comment (pkoro) [inline bin/utils_test.py]: Same comment here

File: conf/EGI_db_conf.json [hunk 4]
```diff
@@ -0,0 +1,20 @@
+{
+  "db_conf": [
+    {
+      "store": "ar",
+      "server": "192.168.0.99",
+      "port": 27017,
+      "database": "argo-egi",
```
Comment (pkoro) [inline conf/EGI_db_conf.json]: Same comment here and once more below (same file)

**Comment by pkoro** [comment]
@kaggis: Should we delete from the [main configuration file][l1] the `mongo_host` and `mongo_port` lines?

[l1]: https://github.com/ARGOeu/argo-compute-engine/blob/ARGO-146-Multitenancy/conf/ar-compute-engine.conf#L4

**Comment by pkoro** [comment]
Although now that I think a bit about it, we could leave these in order to refer to `argo_core` database? Let me know what you think. Maybe I am overthinking here.

**Pull request merged**